homeassistant/util/location.py

```diff
@@ -22,14 +19,16 @@
 
 
 def detect_location_info():
-    """ Detect location information. """
+    """Detect location information."""
     try:
-        raw_info = requests.get(
-            'https://freegeoip.net/json/', timeout=5).json()
+        raw_info = requests.get('https://freegeoip.net/json/', timeout=5)
     except requests.RequestException:
```

Comment:
A better solution would be to catch the JSON decode error too. None of the other changes will be needed.

```python
except (requests.RequestsException, json.decoder.JSONDecodeError):
```